ctapipe/instrument/camera/geometry.py

```diff
@@ -88,25 +88,32 @@ class CameraGeometry:
 
     Parameters
     ----------
-    self : type
-        description
     name : str
-         Camera name (e.g. NectarCam, LSTCam, ...)
-    pix_id : array(int)
+         Camera name (e.g. "NectarCam", "LSTCam", ...)
+    pix_id : np.ndarray[int]
         pixels id numbers
-    pix_x : array with units
+    pix_x : u.Quantity
         position of each pixel (x-coordinate)
-    pix_y : array with units
+    pix_y : u.Quantity
         position of each pixel (y-coordinate)
-    pix_area : array(float)
-        surface area of each pixel, if None will be calculated
-    neighbors : list(arrays)
-        adjacency list for each pixel
-    pix_type : string
+    pix_area : u.Quantity
+        surface area of each pixel
+    pix_type : PixelShape
         either 'rectangular' or 'hexagonal'
-    pix_rotation : value convertable to an `astropy.coordinates.Angle`
-        rotation angle with unit (e.g. 12 * u.deg), or "12d"
-    cam_rotation : overall camera rotation with units
+    pix_rotation : u.Quantity[angle]
+        rotation of the pixels, global value for all pixels
+    cam_rotation : u.Quantity[angle]
+        rotation of the camera. All coordinates will be interpreted
+        to be rotated by this angle with respect to the definition of the
+        frame the pixel coordinates are defined in.
+    neighbors : list[list[int]]
+        adjacency list for each pixel
+        If not given, will be build from the pixel width and distances
```

Comment:
Is this correct? The init skips the neighbor-related stuff if the value is None, but there is code to calculate it, that gets called in the properties. I assume the code in the init only makes sure that any precomputed arrays are proper boolean sparse matrices?